doc: Update WindowsBuild.md

patrickdes opened the pull request:
This pull request contains changes to the WindowsBuild.md documentation file to correct some options to make the compilation on windows work.

@compnerd

compnerd commented on docs/WindowsBuild.md:
```diff
@@ -99,9 +99,11 @@ cmake -B "S:\b\toolchain" -G Ninja -S S:\toolchain\llvm ^
   -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;cmark;swift;lldb;lld" ^
   -DLLVM_EXTERNAL_PROJECTS="cmark;swift" ^
   -DSWIFT_PATH_TO_LIBDISPATCH_SOURCE=S:\toolchain\swift-corelibs-libdispatch ^
+  -DLLVM_PARALLEL_LINK_JOBS=2 ^
```
Hmm, I've found that LLVM is much easier to link than Swift, is there a reason to restrict the LLVM linking?

patrickdes replied:
It comes from the setting in the azure pipeline :

https://github.com/compnerd/swift-build/blob/master/.ci/vs2019-swift-5.2.yml#L146

```
LLVM_OPTIONS: -DLLVM_ENABLE_LIBEDIT=NO *-DLLVM_PARALLEL_LINK_JOBS=2* ...
```

Since the default azure vm build-agent has about 7Gb RAM, which is about the same as I have, I assumed this setting would be right for me.

I think setting a conservative value here, which would work for most people, is better than trying to optimize and have developers having trouble building on their own machine. Since a complete build may take hours, I think it's better to make sure it works the first time.

Maybe we should add a note that, depending on the memory available, this value may be increased to optimize the build time.

If you want, I may do a full build without the LLVM_PARALLEL_LINK_JOBS=2 and see if it works.

patrickdes replied:
Also, it appears that the line :

```
-C S:\windows-swift\cmake\caches\toolchain-5.2.cmake
```

which set the *SWIFT_BUILD_DISABLE_LLDB_PYTHON_SCRIPTS* variable

```
set(SWIFT_BUILD_DISABLE_LLDB_PYTHON_SCRIPTS TRUE CACHE BOOL "" FORCE)
```

is now important or else it produces the error :

```
CMake Error at cmake/modules/LLVMDistributionSupport.cmake:18 (message):
  Specified distribution component 'lldb-python-scripts' doesn't have a target
```

compnerd replied:
Yes, the `lldb-python-scripts` target would be dependent on the python configuration.  That makes sense.  That is rather unfortunate.   For master, the problem comes down to that VS doesn't package the debug libraries for Python.  I'm left wondering if it is worth just indicating that if you are building a debug version of Swift, you should install Python from python.org as the LLDB REPL does require the Python interop.

If you can actually do a test run without `LLVM_PARALLEL_LINK_JOBS=2` it would be great!

patrickdes replied:
I did some research and, it looks like Visual Studio (2019 at least) install the default version of Python. The issue is that, since the process is silent, it install python with default options without giving the user the choice to select *Download debug binaries (requires VS 2015 or later)*.

The good news is that the Python installation is listed in the *Add/Remove Programs* settings and we can *Modify* the installation and download those *debug binaries*. I tried it and, now I see the *python37_d.lib* file.

I will try to do a full build with python enabled to see if it passes. There is hope.

compnerd replied:
Wow, that’s definitely unexpected, but great news!  Would you be so kind as to document that for others?  You are helping make the REPL on Windows easier to get started with :)

patrickdes replied:
LLVM_PARALLEL_LINK_JOBS is definitely needed on pc with  8GB of memory.

```
LINK : fatal error LNK1102: out of memory
[5781/5938] Linking CXX executable bin\lldb-server.exe
```

And, good news, I have been able to compile with LLDB_ENABLE_PYTHON=YES with
the python installation from Visual Studio 2019 (by including the *debug binaries*).
I will document the process later this week.

There is another issue with the compilation but, since it's unrelated to this pull request,
I will start a thread on the forum instead.

patrickdes committed:
doc: Update WindowsBuild.md

Add a section on how to install Python from Visual Studio, including the
debug binaries.
